test/withDataProviders.test.js

```diff
@@ -0,0 +1,145 @@
+/* global test, expect, beforeEach, describe */
+/* eslint-disable no-console */
+import React from 'react'
+import ReactDOM from 'react-dom'
+import {withDataProviders} from '../src/withDataProviders'
+import {compose} from 'redux'
+import {connect} from 'react-redux'
+import {newTestApp, getData, getDataWithCount, GET_DATA_DELAY} from './common'
+import {Promise} from 'bluebird'
+
+beforeEach(() => {
+  getDataWithCount.counter = 0
+})
+
+test('withDataProviders (needed=true) renders child component when the data is fetched', async () => {
+  const {root} = renderMessageContainerApp({needed: true})
+
+  let renderedMessage = root.querySelector('div.message p')
+  expect(renderedMessage).toBeNull()
+
+  await Promise.delay(GET_DATA_DELAY)
```

Comment:
I'd add a safe delay to this, as it's not guaranteed that the promise you are awaiting will be completed *exactly* after the delay, just that it'll be not sooner.

Reply:
I've used jest's `useFakeTimers()` to add more control over all the timings, since I agree that the former approach is unreliable and prone to intermittent test failures.